### ISSUE TITLE
What's the difference between "Dedicated" and "Mobile" graphics in README?

### ISSUE THREAD

**Issue by tevino** [Floris497/mac-pixel-clock-patch-V2#128]
These two columns in the table got me confused:
1. Nvidia **Dedicated** Graphics 10.12 and newer
2. Nvidia **Mobile** Graphics 10.12 and newer

In my case, I got two video cards: `Intel HD Graphics 4000` and `NVIDIA GeForce GT 650M`.

I want to patch for the NVIDIA one, it's a **dedicated** video card obviously, and I suppose the `M` in `650M` stands for `Mobile` thus I got both `Dedicated` and `Mobile`, which one should I take in the table?

**Comment by Floris497** [comment]
At this moment the Nvidia one is not up to date. Please try CoreDisplay and see if it does anything. Else you also need the Nvidia one..

**Comment by tevino** [comment]
@Floris497 
I tried CoreDisplay but it doesn't work:

1080p with 60Hz appears neither in `Displays` nor `SwitchResX`.

I tried this for SwitchResX: http://www.madrau.com/support/support/faq_files/ns_How_can_I_define_a_new_HiDPI_re.html?TB_iframe=true
 
---
 
`NVIDIA-patcher.command` complains:
```
Unknown version of the Nvidia driver found..
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.2 build:16C67

     NVDAGK100Hal: f918269b8c7411f246455376871f7680
 NVDAGK100Hal.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------
```

**Comment by darkfriday** [comment]
i have same issue after try patchd CoreDisplay & NVIDIA.it still not work for me.

```
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.2 build:16C68

     otool CoreDisplay: e9d7a42b6613a45a69a41e8099d0e369
 otool CoreDisplay.bak: e9d7a42b6613a45a69a41e8099d0e369
           CoreDisplay: 91029d061108920340045160051ca109
       CoreDisplay.bak: e843bd7ee58e03aa6e64af71d9e6139c
```
```
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.2 build:16C68

     NVDAGK100Hal: f918269b8c7411f246455376871f7680
 NVDAGK100Hal.bak: NO FILE (this is okay)
```

**Comment by Floris497** [comment]
I will check for an update on the Nvidia one today. getting back to the question from @tevino Dedicated means in this case an PCIe GPU (like a desktop GTX 970) an mobile Nvidia GPU means an Mobile GPU from Nvidia. like the 650M. I will look into changing the description a bit to make it more clear.

**Comment by tevino** [comment]
So glad to hear that. @Floris497 😃

**Comment by Floris497** [comment]
@tevino What OS are you running? I have a possible new patch for 10.12.4 beta. I will check quickly if my other Mac is running 10.12.3 so I can see if it will work for that version as well. for 10.12.2 I can't do anything right now because I don't have any system running this OS.

**Comment by tevino** [comment]
@Floris497 I'm running `10.12.3`, as long as I'm able to backup & restore I'm glad to give it a try.

Did it break your Mac's driver?

**Comment by Floris497** [comment]
Not sure didn't reboot to be honest :p I'm sure it's fine :D if it would fail however I will give you some notes on how to get back to the right version :) anyway.. I'm sure it's not breaking anything

**Comment by tevino** [comment]
@Floris497 I've tried the new version, but still no luck.

**Comment by Floris497** [comment]
Hm, could you try and use IOKit and CoreDisplay at the same time too?

**Comment by Floris497** [comment]
Wait a moment, what cable are you using?

**Comment by tevino** [comment]
Only CoreDisplay could be patched.

NVIDIA:
```
Unknown version of the Nvidia driver found..
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.3 build:16D32

     NVDAGK100Hal: 53b038b824ae3941e7df1f30e41250dc
 NVDAGK100Hal.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------
```
IOKit:
```
Unknown version of IOKit found..
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.3 build:16D32

     otool IOKit: 16edb16d18d52003e010bdcd4070463d
 otool IOKit.bak: NO FILE (this is okay)
           IOKit: df88c0fa7806c6515f0e256f7eec2593
       IOKit.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------
```

---

I use mDP to DP cable, I could change to HDMI if it will work.

**Comment by Floris497** [comment]
No DP is better than HDMI please try IOKIT it has been updated for 10.12.3

**Comment by tevino** [comment]
I was using https://github.com/Floris497/mac-pixel-clock-patch-V2/commit/593a82bad5df5bcd63fe22951b5649e251c36630 which is the latest master until now.

Where can I find the updated IOKIT? Am I missing something?

**Comment by Floris497** [comment]
7 days ago I did the update.. but I'm not running 10.12.3 so I didn't add the MD5's for this one.. please force with last version and check if it did something. if it did could you please provide me the MD5's?

**Comment by tevino** [comment]
Still no luck, Here are what I've done, tell me if I'm doing it wrong:

1. Get the MD5 of unpatched files:
```
$ IOKit-patcher.command md5
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.3 build:16D32

     otool IOKit: 16edb16d18d52003e010bdcd4070463d
 otool IOKit.bak: NO FILE (this is okay)
           IOKit: df88c0fa7806c6515f0e256f7eec2593
       IOKit.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------
```

```
$ CoreDisplay-patcher.command md5
---- BEGINNING MD5 HASH SUMS ---- version: 10.12.3 build:16D32

     otool CoreDisplay: 9c717568024daa81c364a839f09a1bfd
 otool CoreDisplay.bak: NO FILE (this is okay)
           CoreDisplay: a83e2c647a7554f147992488c70ec896
       CoreDisplay.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------
```

2. Make changes so the patch could be applied
```diff
diff --git a/IOKit-patcher.command b/IOKit-patcher.command
index bcf1f34..9619dcf 100755
--- a/IOKit-patcher.command
+++ b/IOKit-patcher.command
@@ -33,6 +33,7 @@ IOKitUnpatched=(
   7359b413a4dca7a189b80da750ce43dd '10.11.1' 6
   # a7afb2dd9df1e4c48f12b4b52f7da212 '10.11.2' 6
   # 3cec8ae287ee52a3622082bfc049bb86 '10.11.3' 6
+  df88c0fa7806c6515f0e256f7eec2593 '10.12.3' 7
 )
 
 # md5 checksums of patched IOKit files
@@ -60,6 +61,7 @@ oToolIOKitUnpatched=(
   21272ae806c3b4ddc593c34ba5d586c0 '10.12 16A254g' 7
   9189a21226c0de8b87c29894f9c374c2 '10.12.1 16B2555' 7
   7feaaa1daf13e4025123fe64dce955c4 '10.12.4 16E154a BETA 1' 7
+  16edb16d18d52003e010bdcd4070463d '10.12.3 16D32' 7
 )
 
 # md5 checksum of '(__DATA,__data)' section exported by otool from patched IOKits
```

3. run `IOKit-patcher.command` and `CoreDisplay-patcher.command` without any parameter
4. Reboot

**Comment by Floris497** [comment]
Please run both at last version see if that works

**Comment by tevino** [comment]
@Floris497 What exactly do you mean by
> run both at last version

I've run both `IOKit-patcher.command` and `CoreDisplay-patcher.command` with [the latest master](593a82bad5df5bcd63fe22951b5649e251c36630), the steps I took are also provided in detail, could you please point out which step is wrong?

**Comment by Floris497** [comment]
Sorry for the late answer, I didn't see correctly what you did. so were you able to patch?

**Comment by tevino** [comment]
@Floris497 Yes, the patch was a success after the change I posted above(step 2), but the resolution(HiDPI 1080P@60Hz) did not showed up.

**Comment by Floris497** [comment]
Did you make custom resolutions?

**Comment by tevino** [comment]
Yes I added a `3840 x 2160` into **Custom Resolutions** but the status is always `Not installed`, and yes I rebooted after adding it, and I tried remove and add it again back and forth every time I apply patch. Also I tried https://github.com/avibrazil/RDM .

Did I missing something?

**Comment by Floris497** [comment]
Save first, and when it says needs reboot you restart.

**Comment by tevino** [comment]
> Save first, and when it says needs reboot you restart.

That's exactly what I've done.
